### ISSUE TITLE
When builder dies while doing a job, the next builder begins to dismantle rather than continue.

### ISSUE THREAD

**Issue by kaboddy** [ldtteam/minecolonies#7180]
### Prerequisites 

- [ ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [X ] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [X ] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version:  1.15.5
- MineColonies Version:  0.14.68-ALPHA
- Structurize Version: 0.13.148-ALPHA
- Related Mods and their Versions: Valelsia 16.0.6  (3.2.3)

## Expected behavior

We built a jungle village.  When the builder dies while working on a job, we'd expect the auto assigned builder to continue where the other left off...

## Actual behavior

When the builder dies, the auto assigned replacement builder begins to dismantle the current build project.

## Steps to reproduce the issue

I guess build a Jungle Village, build builder hut, start building another building and kill the builder to see what happens.

1. 
2. 
3. 

### Logs

- latest.log: 
- crashlog: 

### Notes
 WorkOrderBuild at location BlockPos{x=-2801, y=74, z=-1375} is using schematics/jungle/university1 instead of schematics/jungle/university1

Found this on console... 

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by XxN4BxX** [comment]
this still happends in minecraft 1.16.5 with minecolonies 0.14.219-ALPHA, not a full dismantle and not just with university, it happends with other buildings as well(not as often but every big building schematics have at least 1 builder die to it), not just that, there are alot of deads when building with jungle designs, deads are mostly related to the fact that builders have to place multiple vines on the buildings themselves, vines should only be needed on the top most block, instead of the column they have to build since they grow themselves over time, when one dies and the other takes oves, if there are vines that have grown they will always trim the vines before starting to build again

**Comment by Raycoms** [comment]
Yeah, happens in all level 0 buildings, since the progress is reset on death.

**Comment by McSquizzleswag** [comment]
I'm experiencing this in my 1.18.2 server, with a builder dismantling a warehouse, is the best course of action to just let the second builder tear down the warehouse and rebuild it?

**Comment by Raycoms** [comment]
they will not tear it down entirely, only a small part of it.

**Comment by McSquizzleswag** [comment]
I didn't stick around long enough to see what happened, I had went and checked on the status and saw the torches were placed, I thought "finally it's done," but saw the builder removing some walls and then I left thinking maybe the tick timer being disabled had something to do with it

**Comment by Mekadrom** [comment]
From my experience, only Domum Ornamentum blocks are removed by the next builder, and they appear to skip vanilla Minecraft blocks. Perhaps something about the way Domum stores block states is failing some validation the builder does as soon as they assume the position?

**Comment by uecasm** [comment]
It happens with any fence/stairs-like block (including DO shingles, since they work like stairs), because their state depends on the adjacent blocks, so if all the adjacent blocks aren't placed yet then at least one of them will be "wrong".  So they remove that one, and now another one is "wrong", etc.  They'll eventually sort it out, as long as you keep them from dying.
